### PR TITLE
repos: add 5 new endpoints relating to issues

### PR DESCRIPTION
**Pull request by AlexNPavel**
This adds new endpoints for issues, mainly for issue comments (PR
comments are accessed via the Issues Comments API as well).

All 5 of the new endpoints have been tested.

### CODE REVIEW

File: src/repos/get.rs [hunk 1]
```diff
@@ -50,6 +55,24 @@ from!(
        => CommitsSha
     @Commits
        => Executor
+    @Issues
+       -> IssuesComments = "comments"
+    @Issues
+       => IssuesNumber
+    @Issues
```
Comment (mgattozzi) [inline src/repos/get.rs]: If you look here:
https://github.com/mgattozzi/github-rs/blob/11d9cf7fa0f3f90a54b78f61df3566fc128c2b0f/src/users/get.rs#L75-L77
you can see that you can have multiple different types assigned so you don't need to do `@Issues` more than once! It's a subtle part of the macro that unfortunately I haven't documented as well.

File: src/repos/get.rs [hunk 2]
```diff
@@ -50,6 +55,24 @@ from!(
...
+       => Executor
+    @IssuesComments
+       => IssuesCommentsId
+    @IssuesComments
```
Comment (mgattozzi) [inline src/repos/get.rs]: Same as above with all of these!

File: src/repos/get.rs [hunk 3]
```diff
@@ -137,7 +161,31 @@ impl_macro!(
         |=> sha -> CommitsSha = sha_str
     @Commits
         |
-        |->execute
+        |-> execute
+
+    @Issues
+        |
+        |=> number -> IssuesNumber = issue_number
+    @Issues
```
Comment (mgattozzi) [inline src/repos/get.rs]: You can combine this with the one above it

File: src/repos/get.rs [hunk 4]
```diff
@@ -137,7 +161,31 @@ impl_macro!(
...
+        |=> comments -> IssuesComments
+        |
+        |-> execute
+    @IssuesComments
+        |
+        |=> id -> IssuesCommentsId = comment_id
+    @IssuesComments
```
Comment (mgattozzi) [inline src/repos/get.rs]: Same with this one

**Comment by AlexNPavel** [comment]
Thanks. This is one of the cases where some macro documentation could help for the impl_macro. The different parts need to be in a certain order otherwise it breaks. I was able to figure it out today. My internship ends this week, so I can look over the design docs next week and add some documentation for this macro.

**Comment by mgattozzi** [comment]
@AlexNPavel that would be great! I wish I could make the syntax a bit cleaner so I just need to fix the macro at some point but it ends up with these "unspecific enough match" problems.

**Pull request merged**